src/libstd/net/tcp.rs

```diff
@@ -130,8 +130,13 @@ impl TcpStream {
     /// If the value specified is `None`, then `read` calls will block
     /// indefinitely. It is an error to pass the zero `Duration` to this
     /// method.
-    #[unstable(feature = "socket_timeout", reason = "RFC 1047 - recently added",
-               issue = "27773")]
+    ///
+    /// # Note
+    ///
+    /// Platforms may return a different error code whenever a read times out as
+    /// a result of setting this option. For example Unix typically returns an
+    /// error of the kind `WouldBlock`, but Windows may return `TimedOut`.
+    #[stable(feature = "socket_timeout", since = "1.4.0")]
```

Comment:
To elaborate on this a bit: we discussed #27773 extensively in the libs team meeting, and I've written up the results of that discussion [in a new comment there](https://github.com/rust-lang/rust/issues/27773#issuecomment-139604754).